apps/hash-api/src/graphql/resolvers/ontology/data-type.ts

```diff
@@ -41,17 +44,17 @@ export const getDataType: ResolverFn<
   {},
   GraphQLContext,
   QueryGetDataTypeArgs
-> = async (_, { dataTypeId, constrainsValuesOn }, { dataSources, user }) => {
-  const context = dataSourcesToImpureGraphContext(dataSources);
-
-  return await getDataTypeSubgraphById(context, {
-    dataTypeId,
-    actorId: user ? user.accountId : systemUserAccountId,
-    /** @todo - make these configurable once non-primitive data types are a thing https://app.asana.com/0/1200211978612931/1202464168422955/f */
-    graphResolveDepths: {
-      ...zeroedGraphResolveDepths,
-      constrainsValuesOn,
+> = async (_, { dataTypeId, constrainsValuesOn }, { dataSources, user }) =>
+  getDataTypeSubgraphById(
+    dataSourcesToImpureGraphContext(dataSources),
+    { actorId: user?.accountId ?? publicUserAccountId },
```

Comment:
I don't think we want to have to remember to do this for any request that can be made by a public person – that means we have two places where we are tracking 'this action is public' instead of one central permission store.

I suggest we add `actorId` to the context in `create-apollo-server.ts`:
```ts
  return new ApolloServer({
    schema: combinedSchema,
    dataSources: getDataSources,
    context: (ctx): Omit<GraphQLContext, "dataSources"> => ({
      ...ctx,
      actorId: ctx.req.user?.accountId ?? publicUserAccountId,
      user: ctx.req.user,
      // etc
```
then we can use `actorId` from the context in all the resolvers, rather than having to make any decisions about which should have public and which not.

Reply:
I think that's a good point. Will do the changes. thank you!

Reply:
I moved the context in https://github.com/hashintel/hash/pull/3103/commits/30f4213dafb4318a229985fe0d83d7dbe3fe7656

Reply:
Great stuff!